tests/introspection/ApiTypoTest.cs

```diff
@@ -713,9 +716,46 @@ public void TypoTest ()
 
 					var methods = t.GetMethods ();
 					foreach (MethodInfo m in methods) {
-					if (!m.IsPublic && !m.IsFamily && !IsObsolete (m))
+						if (!m.IsPublic && !m.IsFamily)
+								continue;
+
+						// Availability attribute's message checks.
+						var ca = m.GetCustomAttributes<AvailabilityBaseAttribute> ();
+						if (ca.Any ()) {
```

Comment:
You can remove `if (ca.Any ())`, it's redundant since the foreach will just not iterate over anything if the list is empty.

This will also reduce the indentation level for all the code :smile: